pycbc/inference/models/marginalized_gaussian_noise.py

```diff
@@ -194,6 +194,157 @@ def _loglr(self):
+                wfs = self.waveform_generator.generate(**params)
+            else:
+                wfs = {}
+                for det in self.data:
+                    wfs.update(self.waveform_generator[det].generate(**params))
+        except NoWaveformError:
+            return self._nowaveform_loglr()
+        except FailedWaveformError as e:
+            if self.ignore_failed_waveforms:
+                return self._nowaveform_loglr()
+            else:
+                raise e
+
+        sh_total = hh_total = 0.
+        snr_estimate = {}
+        cplx_hpd = {}
+        cplx_hcd = {}
+        hphp = {}
+        hchc = {}
+        hphc = {}
+        for det, (hp, hc) in wfs.items():
+            # the kmax of the waveforms may be different than internal kmax
+            kmax = min(max(len(hp), len(hc)), self._kmax[det])
+            slc = slice(self._kmin[det], kmax)
+
+            # whiten both polarizations
+            hp[self._kmin[det]:kmax] *= self._weight[det][slc]
+            hc[self._kmin[det]:kmax] *= self._weight[det][slc]
+
+            hp.resize(len(self._whitened_data[det]))
+            hc.resize(len(self._whitened_data[det]))
+            cplx_hpd[det], _, _ = matched_filter_core(
+                                 hp,
+                                 self._whitened_data[det],
+                                 low_frequency_cutoff=self._f_lower[det],
+                                 high_frequency_cutoff=self._f_upper[det],
+                                 h_norm=1)
+            cplx_hcd[det], _, _ = matched_filter_core(
+                                 hc,
+                                 self._whitened_data[det],
+                                 low_frequency_cutoff=self._f_lower[det],
+                                 high_frequency_cutoff=self._f_upper[det],
+                                 h_norm=1)
+
+            hphp[det] = hp[slc].inner(hp[slc]).real
+            hchc[det] = hc[slc].inner(hc[slc]).real
+            hphc[det] = hp[slc].inner(hc[slc]).real
+
+            snr_proxy = ((cplx_hpd[det] / hphp[det] ** 0.5).squared_norm() +
+                         (cplx_hcd[det] / hchc[det] ** 0.5).squared_norm())
+            snr_estimate[det] = (0.5 * snr_proxy) ** 0.5
+
+        self.snr_draw(snr_estimate)
+
+        for det in wfs:
+            if det not in self.dets:
+                self.dets[det] = Detector(det)
+            fp, fc = self.dets[det].antenna_pattern(
+                                    params['ra'],
+                                    params['dec'],
+                                    params['polarization'],
+                                    params['tc'])
+            dt = self.dets[det].time_delay_from_earth_center(params['ra'],
+                                                             params['dec'],
+                                                             params['tc'])
+            dtc = params['tc'] + dt
+            cplx_hd = fp * cplx_hpd[det].at_time(dtc,
```

Comment:
@cdcapano This is where the sub-sample interpolation comes into play. The monte-carlo points for the time are not limited to the sample locations. If you recall, the times are choices by their sample, but then we dither the time to +- half a sample, so all times are covered.